src/main/java/RacingGame.java

```diff
@@ -0,0 +1,52 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RacingGame {
+    private List<RacingCar> racingCars = new ArrayList<>();
+    private final Random random;
+    private int runCount;
+
+    public RacingGame(Random random){
+        this.random =  random;
+    }
+
+    public void initializeGame(String[] carNames, int runCount) {
+        this.runCount = runCount;
+        for (String name : carNames) {
+            racingCars.add(new RacingCar(name));
+        }
+    }
+
+    public void playRound() {
+        for (RacingCar car : racingCars) {
+            car.move(random.nextInt(10));
+        }
+    }
+
+    public int getRunCount() {
+        return runCount;
+    }
+
+    public List<RacingCar> getRacingCars() {
+        return racingCars;
+    }
+
+    public List<String> findWinners() {
+        int maxPosition = racingCars.stream()
+                .mapToInt(RacingCar::getForwardCount)
+                .max()
+                .orElse(0);
+        List<String> winners = new ArrayList<>();
+        for (RacingCar car : racingCars) {
+            checkWinners(car, maxPosition, winners);
+        }
+        return winners;
+    }
+
+    private void checkWinners(RacingCar car, int maxPosition, List<String> winners){
+        if (car.getForwardCount() == maxPosition) {
+            winners.add(car.getCarName());
+        }
+    }
```

Comment:
void함수에서 인자의 참조를 이용해서 winners에 자동차를 추가하고 있어요!
checkWinners라는 메소드의 이름과 다르게 인자의 값을 바꿔버리는 예상치 못한 동작을 할 수 있을 것 같습니다.
(다른 사람이 해당 함수를 사용할 때 우승자인지 검사할줄 알았는데 인자로 넘겨준 winners가 바뀌면 당황스럽겠죠?)

이런 경우에는 checkWinner라는 함수에서 boolean을 반환하고, 우승자가 맞다면 winners에 추가해주는것도 좋아보입니다!
```suggestion
        List<String> winners = new ArrayList<>();
        for (RacingCar car : racingCars) {
            if (checkWinner(car, maxPosition)) {
                winners.add(car.getCarName);
            }
        }
        return winners;
    }

    private boolean checkWinner(RacingCar car, int maxPosition){
        return car.getForwardCount() == maxPosition;
    }
```

Reply:
depth 1길이에 맞춰 모듈화를 하다보니 예상치 못한 실수를 한것 같습니다... 꼼꼼히 봐주셔서 감사합니다! 머리로는 알고 있어도 막상 코드를 짜보니 boolean을 통한 검사는 생각치 못했네요..